diff --git a/src/Makefile b/src/Makefile
index b771196..0933de5 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -6,13 +6,14 @@
 # Your platform. See PLATS for possible values.
 PLAT= guess
 
-CC= gcc -std=gnu99
+CC?= gcc -std=gnu99
 CFLAGS= -O2 -Wall -Wextra -DLUA_COMPAT_5_3 $(SYSCFLAGS) $(MYCFLAGS)
 LDFLAGS= $(SYSLDFLAGS) $(MYLDFLAGS)
 LIBS= -lm $(SYSLIBS) $(MYLIBS)
 
-AR= ar rcu
-RANLIB= ranlib
+AR?= ar
+AR_CALL= $(AR) rcu
+RANLIB?= ranlib
 RM= rm -f
 UNAME= uname
 
@@ -57,7 +58,7 @@ o:	$(ALL_O)
 a:	$(ALL_A)
 
 $(LUA_A): $(BASE_O)
-	$(AR) $@ $(BASE_O)
+	$(AR_CALL) $@ $(BASE_O)
 	$(RANLIB) $@
 
 $(LUA_T): $(LUA_O) $(LUA_A)
@@ -81,7 +82,7 @@ echo:
 	@echo "CFLAGS= $(CFLAGS)"
 	@echo "LDFLAGS= $(LDFLAGS)"
 	@echo "LIBS= $(LIBS)"
-	@echo "AR= $(AR)"
+	@echo "AR= $(AR_CALL)"
 	@echo "RANLIB= $(RANLIB)"
 	@echo "RM= $(RM)"
 	@echo "UNAME= $(UNAME)"
